### ISSUE TITLE
Fix go test call in Makefile

### ISSUE THREAD

**Issue by jodh-intel** [kata-containers/kata-containers#1494]
The `go-test` rule in the runtime makefile is incorrect - it is calling `go test` directly:

https://github.com/kata-containers/kata-containers/blob/main/src/runtime/Makefile#L638..L639

What it *should* be doing is calling [`ci/go-test.sh`](https://github.com/kata-containers/kata-containers/blob/main/ci/go-test.sh) since that script:

- Call the unit tests *twice* if possible (once as non-`root` and once as `root`).
- Merges the results.
- Resets file permissions.

In summary, the 2.x runtime makefile should look more like the 1.x runtime makefile:

- https://github.com/kata-containers/runtime/blob/master/Makefile#L658..L659

**Comment by Bevisy** [comment]
@jodh-intel pls take a look #2481
